test/unit/project_type/test_git.py

```diff
@@ -286,6 +286,31 @@ def fake_popen_constructor(command, stdout, stderr, *args, **kwargs):
         project_type_popen_patch.side_effect = fake_popen_constructor
         return project_type_popen_patch
 
+    def test_get_repo_directory_with_no_netloc(self):
+        Configuration['repo_directory'] = join(expanduser('~'), '.clusterrunner', 'repos')
+        url = 'git.dev.box.net:Productivity/ClusterRunnerHealthCheck'
+
+        actual_repo_directory = Git.get_full_repo_directory(url)
+        expected_repo_directory = join(
+            Configuration['repo_directory'],
+            'git.dev.box.net',
+            'Productivity',
+            'ClusterRunnerHealthCheck',
+        )
+        self.assertEqual(expected_repo_directory, actual_repo_directory)
+
+    def test_get_repo_directory_with_netloc(self):
+        Configuration['repo_directory'] = join(expanduser('~'), '.clusterrunner', 'repos')
+        url = 'ssh://scm.dev.box.net:12345/awesome-project'
+
+        actual_repo_directory = Git.get_full_repo_directory(url)
+        expected_repo_directory = join(
+            Configuration['repo_directory'],
+            'scm.dev.box.net12345',
+            'awesomeproject',
+        )
+        self.assertEqual(expected_repo_directory, actual_repo_directory)
```

Comment:
These look like good tests to combine into a single genty-driven test. You could also probably combine it with the `test_get_full_repo_directory` above.

If you search the test directory for genty you'll find examples. [Here's one.](https://github.com/box/ClusterRunner/blob/0d2f5b54/test/integration/common/test_console_output.py#L38-L56)

I like to combine similar tests like that because it reduces duplication, but more importantly it makes it easier to see that these are really the same test just with different inputs.